k8s/configuration/secret.yaml

```diff
@@ -0,0 +1,92 @@
+# All secret data should be base64 encoded before embedding them here by
+# using `echo "secret string" | base64 -w 0 > secret.string.b64` and then
+# copy the resulting value here.
+# Ref: https://kubernetes.io/docs/concepts/configuration/secret/
+
+apiVersion: v1
+kind: Secret
+metadata:
+  name: mdb-agent-api-key
+  namespace: default
+type: Opaque
+data:
+  # This is the API Key obtained from MongoDB Cloud Manager
+  api-key: "<b64 encoded api key>"
+---
+apiVersion: v1
+kind: Secret
+metadata:
+  name: bdb-private-key
+  namespace: default
+type: Opaque
+data:
+  private.key: "<b64 encoded private key>"
+---
+apiVersion: v1
+kind: Secret
+metadata:
+  name: mdb-certs
+  namespace: default
+type: Opaque
+data:
+  mdb-instance.pem: "<b64 encoded, concatanated public and private keys>"
+  ca.pem: "<b64 encoded CA public key>" 
```

Comment:
`ca.pem` is required by almost all the sub-systems that make up a node.
The reason I put it many times is so that we can isolate secret access in between pods. That is, a pod that has access for secret 1, cannot peep into secret 2.

What you are proposing is to have a common secret `ca` which each pod can access.
That should be possible. I propose that we handle it in a separate PR as it is a usability enhancement. Sounds good?